exist-core/src/main/java/org/exist/repo/ClasspathHelper.java

```diff
@@ -169,7 +170,18 @@ private static void scanPackageDir(Classpath classpath, Path module) throws IOEx
                 try (final BufferedReader reader = Files.newBufferedReader(cp)) {
                     String line;
                     while ((line = reader.readLine()) != null) {
-                        classpath.addComponent(line);
+                        Path p = Paths.get(line);
+                        if (!p.isAbsolute()) {
+                            final FileSystemStorage.FileSystemResolver res = (FileSystemStorage.FileSystemResolver) pkg.getResolver();
+                            p = res.resolveComponentAsFile(line);
+                        }
+                        p = p.normalize().toAbsolutePath();
+
+                        if (!Files.exists(p)) {
+                            LOG.warn("Unable to add '" + p + "' to the classpath for EXPath package: " + pkg.getName() + ", as the file does not exist!");
+                        } else {
+                            classpath.addComponent(p.toString());
+                        }
```

Comment:
it may would be easier to read if you turn the if statement and remove the inversion within the if clause like so:
```
 if (Files.exists(p)) {
     classpath.addComponent(p.toString());
 } else {
     LOG.warn("Unable to add '" + p + "' to the classpath for EXPath package: " + pkg.getName() + ", as the file does not exist!");
 }
 ```